src/Service/U2fService.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupU2fBundle\Service;
+
+use Surfnet\StepupU2fBundle\Dto\RegisterRequest;
+use Surfnet\StepupU2fBundle\Dto\RegisterResponse;
+use Surfnet\StepupU2fBundle\Dto\Registration;
+use Surfnet\StepupU2fBundle\Dto\SignRequest;
+use Surfnet\StepupU2fBundle\Dto\SignResponse;
+use Surfnet\StepupU2fBundle\Exception\LogicException;
+use Surfnet\StepupU2fBundle\Value\AppId;
+use u2flib_server\Error;
+use u2flib_server\RegisterRequest as YubicoRegisterRequest;
+use u2flib_server\Registration as YubicoRegistration;
+use u2flib_server\SignRequest as YubicoSignRequest;
+use u2flib_server\U2F;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects) — Mainly due to the DTOs.
+ */
+class U2fService
+{
+    /**
+     * @var \u2flib_server\U2F
+     */
+    private $u2fService;
+
+    /**
+     * @var AppId
+     */
+    private $appId;
+
+    /**
+     * @param AppId $appId
+     * @param U2F    $u2fService
+     */
+    public function __construct(AppId $appId, U2F $u2fService)
+    {
+        $this->appId = $appId;
+        $this->u2fService = $u2fService;
+    }
+
+    /**
+     * @return RegisterRequest
+     */
+    public function requestRegistration()
```

Comment:
`createRegistrationRequest` or something seems more appropriate as it creates a registration request rather than request a registration?